packages/assistify-ai/server/hooks/sendMessageToKnowledgeAdapter.js

```diff
@@ -1,43 +1,76 @@
 /* globals RocketChat, SystemLogger */
 
-import { getKnowledgeAdapter } from '../lib/KnowledgeAdapterProvider';
+import {getKnowledgeAdapter} from '../lib/KnowledgeAdapterProvider';
 
 RocketChat.callbacks.remove('afterSaveMessage', 'externalWebHook');
 
-RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
-	// skips this callback if the message was edited
-	if (message.editedAt) {
-		return message;
-	}
-
+function isMessageRelevant(message, room) {
 	let knowledgeEnabled = false;
 	RocketChat.settings.get('Assistify_AI_Enabled', function(key, value) {
 		knowledgeEnabled = value;
 	});
 
 	if (!knowledgeEnabled) {
-		return message;
+		return false;
+	}
+
+	//we only want to forward messages from livechat-rooms - requests are implemented in the help-request-package
+	if (!room) {
+		room = RocketChat.models.Rooms.findOneById(message.rid);
 	}
 
-	//we only want to forward messages from livechat-rooms
 	if (!(room && (room.t === 'l'))) {
-		return message;
+		return false;
 	}
 
 	const knowledgeAdapter = getKnowledgeAdapter();
 	if (!knowledgeAdapter) {
-		return;
+		return false;
 	}
 
-	SystemLogger.debug(`Send message ${ message._id } to knowledgeAdapter (Meteor.defer()`);
+
+}
+
+RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
+	if (isMessageRelevant(message, room)) {
```

Comment:
This function ```isMessageRelevant``` returns **false** when certain conditions fail, should it need to explicitly return **true** if the message is relevant?